src/reflect/value.go

```diff
@@ -355,13 +352,16 @@ func (v Value) Len() int {
 	case Array:
 		return v.typecode.Len()
 	case Chan:
-		return chanlen(v.value)
+		return chanlen(v.pointer())
 	case Map:
-		return maplen(v.value)
+		return maplen(v.pointer())
 	case Slice:
-		return int((*sliceHeader)(v.value).len)
+		if v.pointer() == nil {
+			return 0
+		}
+		return int((*sliceHeader)(v.pointer()).len)
 	case String:
-		return int((*stringHeader)(v.value).len)
+		return int((*stringHeader)(v.pointer()).len)
```

Comment:
Same here.
Also, if slices can have a nil value, then strings probably as well. They are very similar as far as reflect is concerned. (I need to think this through a bit more when I'm not tired).